pkg/ioutils/readers.go

```diff
@@ -74,14 +93,61 @@ func NewBufReaderWithDrainbufAndBuffer(r io.Reader, drainBuffer []byte, buffer *
 }
 
 func (r *bufReader) drain() {
+	var (
+		duration       time.Duration
+		lastReset      time.Time
+		now            time.Time
+		reset          bool
+		bufLen         int64
+		dataSinceReset int64
+		maxBufLen      int64
+		reuseBufLen    int64
+		reuseCount     int64
+	)
+	reuseBufLen = int64(len(r.reuseBuf))
```

Comment:
Also I'm not sure about conversion all to `int64`. We operating numbers not even nearly as great as MAXINT on 32-bit systems. Why we need complicate code here?